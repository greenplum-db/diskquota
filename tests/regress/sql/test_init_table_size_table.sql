-- heap table
CREATE TABLE t(i int);
INSERT INTO t SELECT generate_series(1, 100000);

-- heap table index
CREATE INDEX idx on t(i);

-- toast table
CREATE TABLE toast(t text);
INSERT INTO toast SELECT repeat('a', 10000) FROM generate_series(1, 1000);

-- toast table index
CREATE INDEX toast_idx on toast(t);

-- AO table
CREATE TABLE ao (i int) WITH (appendonly=true);
INSERT INTO ao SELECT generate_series(1, 100000);

-- AO table index
CREATE INDEX ao_idx on ao(i);

-- AOCS table
CREATE TABLE aocs (i int, t text) WITH (appendonly=true, orientation=column);
INSERT INTO aocs SELECT i, repeat('a', 1000) FROM generate_series(1, 10000) AS i;

-- AOCS table index
CREATE INDEX aocs_idx on aocs(i);

SELECT pg_sleep(5);

select tableid::regclass, size from diskquota.table_size where segid = -1 and tableid::regclass::name not like '%.%' order by tableid;

-- init diskquota.table_size
SELECT diskquota.init_table_size_table();
select tableid::regclass, size from diskquota.table_size where segid = -1 and tableid::regclass::name not like '%.%' order by tableid;

DROP TABLE t;
DROP TABLE toast;
DROP TABLE ao;
DROP TABLE aocs;
