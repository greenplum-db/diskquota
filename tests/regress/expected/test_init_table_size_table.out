-- heap table
CREATE TABLE t(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO t SELECT generate_series(1, 100000);
-- heap table index
CREATE INDEX idx on t(i);
-- toast table
CREATE TABLE toast(t text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 't' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO toast SELECT repeat('a', 10000) FROM generate_series(1, 1000);
-- toast table index
CREATE INDEX toast_idx on toast(t);
-- AO table
CREATE TABLE ao (i int) WITH (appendonly=true);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO ao SELECT generate_series(1, 100000);
-- AO table index
CREATE INDEX ao_idx on ao(i);
-- AOCS table
CREATE TABLE aocs (i int, t text) WITH (appendonly=true, orientation=column);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO aocs SELECT i, repeat('a', 1000) FROM generate_series(1, 10000) AS i;
-- AOCS table index
CREATE INDEX aocs_idx on aocs(i);
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

select tableid::regclass, size from diskquota.table_size where segid = -1 and tableid::regclass::name not like '%.%' order by tableid;
  tableid  |   size   
-----------+----------
 t         |  3932160
 idx       |  2490368
 toast     |   393216
 toast_idx |   327680
 ao        |  1591464
 ao_idx    |  2490368
 aocs      | 10813592
 aocs_idx  |   524288
(8 rows)

-- init diskquota.table_size
SELECT diskquota.init_table_size_table();
 init_table_size_table 
-----------------------
 
(1 row)

select tableid::regclass, size from diskquota.table_size where segid = -1 and tableid::regclass::name not like '%.%' order by tableid;
  tableid  |   size   
-----------+----------
 t         |  3932160
 idx       |  2490368
 toast     |   393216
 toast_idx |   327680
 ao        |  1591464
 ao_idx    |  2490368
 aocs      | 10813592
 aocs_idx  |   524288
(8 rows)

DROP TABLE t;
DROP TABLE toast;
DROP TABLE ao;
DROP TABLE aocs;
